apps/condo/domains/notification/adapters/hcmAdapter.js

```diff
@@ -0,0 +1,316 @@
+const {
+    EMPTY_HCM_CONFIG_ERROR,
+    INVALID_HCM_CONFIG_ERROR,
+    EMPTY_NOTIFICATION_TITLE_BODY_ERROR,
+} = require('@condo/domains/notification/constants/errors')
+
+const {
+    DEFAULT_NOTIFICATION_OPTIONS,
+    PUSH_SUCCESS_CODE, PUSH_PARTIAL_SUCCESS_CODE, SUCCESS_CODES,
+} = require('./hcm/constants')
+
+const HCMMessaging = require('./hcm/messaging')
+
+const HCM_CONFIG = conf[HCM_CONFIG_ENV] ? JSON.parse(conf[HCM_CONFIG_ENV]) : null
+const DEFAULT_PUSH_SETTINGS = {}
+const CONFIG_VALIDATED_FIELDS = [APP_MASTER_KEY, APP_RESIDENT_KEY, `${APP_MASTER_KEY}.clientId`, `${APP_MASTER_KEY}.secret`, `${APP_RESIDENT_KEY}.clientId`, `${APP_RESIDENT_KEY}.secret`]
+const IS_TEST_ENV = conf.NODE_ENV === 'test'
+const IS_LOCAL_ENV = conf.SERVER_URL.includes('localhost')
+
+const logger = getLogger('HCMAdapter')
+
+const getRandomDigit = () => Math.floor(Math.random() * 10)
+const getRandomNumeric = (count) => {
+    const result = []
+
+    while (count > 0 && result.length < count) result.push(getRandomDigit())
+
+    return result.join('')
+}
+
+/**
+ * HCM is Huawei Cloud Messaging
+ *
+ * Send push notification to pushToken via app, configured by HCM_CONFIG in .helm (.env)
+ * Attention! Notifications could only be sent to devices, connected via same PROJECT_IDs.
+ * Attempts to send push notifications to devices, connected through different projects will fail.
+ */
+class HCMAdapter {
+    apps = {
+        [APP_RESIDENT_KEY]: null,
+        [APP_MASTER_KEY]: null,
+    }
+
+    constructor (config = HCM_CONFIG) {
+        try {
+            HCMAdapter.validateConfig(config)
+
+            this.apps[APP_MASTER_KEY] = new HCMMessaging(config[APP_MASTER_KEY])
+            this.apps[APP_RESIDENT_KEY] = new HCMMessaging(config[APP_RESIDENT_KEY])
+        } catch (error) {
+
+            // For CI/local tests config is useless because of emulation via FAKE tokens
+            logger.error({ msg: 'HCMAdapter error', error })
+        }
+    }
+
+    /**
+     * Return app type (resident/master) based on appId detected by token
+     * @param appIds
+     * @param token
+     * @returns {*}
+     */
+    static getAppType (appIds, token) {
+        return HUAWEI_APP_TYPE_BY_APP_ID[appIds[token]]
+    }
+
+    /**
+     * Validates HCM config
+     * @param config
+     */
+    static validateConfig (config) {
+        if (isEmpty(config)) throw new Error(EMPTY_HCM_CONFIG_ERROR)
+
+        CONFIG_VALIDATED_FIELDS.forEach(
+            (field) => {
+                if (isEmpty(get(config, field))) throw new Error(INVALID_HCM_CONFIG_ERROR)
+            }
+        )
+    }
+
+    /**
+     * Converts all non-string fields to string
+     * @param data
+     * @param token
+     * @returns {{token}}
+     */
+    static prepareData (data = {}, token) {
+        const result = { token }
+
+        for (const key in data) {
+            if (data.hasOwnProperty(key)) result[key] = isString(data[key]) ? data[key] : String(data[key])
```

Comment:
That's strange) I think it's not need for check for result param is string or not. `String` over `string` variable also gives you a string type. So you can simplify this line a little
```suggestion
            if (data.hasOwnProperty(key)) result[key] = String(data[key])
```